src/components/Button/Button.tsx

```diff
@@ -0,0 +1,68 @@
+﻿import classNames from 'classnames';
+
+import scss from '~components/Button/Button.module.scss';
+
+import { ITextButtonProps, ICircleButtonProps } from './Button.types';
+
+function Button(props: ITextButtonProps | ICircleButtonProps) {
+  const RenderTextButton = () => {
+    const {
+      disableElevation,
+      className,
+      children,
+      disabled,
+      onClick,
+      variant,
+      color,
+      type,
+    } = props as ITextButtonProps;

```

Comment:
a gente não usa desestruturação de props no projeto, somente em casos específicos, se isso não for algo crítico teria como você mudar isso?